### PR TITLE
Refactor to TypeScript

### PR DESCRIPTION
**Pull request by Hoishin**
- Refactored to TypeScript
- Uses `@typescript-eslint/parser` for eslint
- Update some packages to line up with its type definition

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
+	"parser": "@typescript-eslint/parser",
 	"extends": "xo/esnext",
-	"ecmaFeatures": {
-		"globalReturn": true
+	"rules": {
+		"no-unused-vars": "off",
```
Comment (Hoishin) [inline .eslintrc:5]: We don't have to check this since TypeScript compiler does this.

Comment (Lange) [reply .eslintrc]: If porting to TypeScript, eslint should be removed and replaced with tslint.

Comment (Hoishin) [reply .eslintrc:5]: Recently Microsoft decided to throw away TSLint and focus on support on ESLint.  `@typescript-eslint` project is among that.

Comment (Hoishin) [reply .eslintrc:5]: Reference: https://github.com/Microsoft/TypeScript/issues/29288

Comment (Lange) [reply .eslintrc]: Oh, interesting. I was unaware

Comment (Hoishin) [reply .eslintrc:5]: Yeah, I was trying to put this info in Discord but been distracted.

**Comment by Hoishin** [comment]
Once this is merged, I will publish prerelease (dist-tag as `next`) to make sure everything works as CLI, then publish as major version.

Does that sound good @Lange ?

**Comment by Lange** [comment]
Sounds good to me! @Hoishin

**Pull request merged**